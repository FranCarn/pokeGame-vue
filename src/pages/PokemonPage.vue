<template>
  <h1 v-if="!pokemon">Loading...</h1>

  <div v-else>
    <h1>Who's that pokemon?</h1>

    <PokemonPicture :pokemonId="pokemon.id" :showPokemon="showPokemon" />
    <PokemonOptions :pokemons="pokemonsArr" @selection="checkAnswer" />
  </div>
</template>

<script>
import PokemonOptions from "@/components/PokemonOptions";
import PokemonPicture from "@/components/PokemonPicture";
import getPokemonOptions from "@/helpers/getPokemonOptions";
export default {
  name: "PokemonPage",
  components: {
    PokemonOptions,
    PokemonPicture,
  },
  data() {
    return {
      pokemonsArr: [],
      pokemon: null,
      showPokemon: false,
    };
  },
  methods: {
    async mixPokemonArray() {
      this.pokemonsArr = await getPokemonOptions();
      const rndInt = Math.floor(Math.random() * 4);
      this.pokemon = this.pokemonsArr[rndInt];
    },
    checkAnswer(id) {
      this.showPokemon = true;
      console.log(id);
    },
  },
  mounted() {
    this.mixPokemonArray();
  },
};
</script>
