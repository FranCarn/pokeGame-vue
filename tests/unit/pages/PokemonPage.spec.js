import { shallowMount } from "@vue/test-utils";
import PokemonPage from "@/pages/PokemonPage";
import { pokemons } from "../mocks/pokemons.mock";
describe("PokemonPage tests", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(PokemonPage);
  });

  test("should match with snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("should call mixPokemonArray when component mount", () => {
    const mixPokemonArraySpy = jest.spyOn(
      PokemonPage.methods,
      "mixPokemonArray"
    );
    shallowMount(PokemonPage);
    expect(mixPokemonArraySpy).toBeCalledTimes(1);
  });

  test("should match with snapshot when pokemons exists", async () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          message: null,
          pokemon: pokemons[0],
          pokemonsArr: pokemons,
          showAnswer: false,
          showPokemon: false,
        };
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test("should show PokemonPicture & PokemonOptions components", () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          message: null,
          pokemon: pokemons[0],
          pokemonsArr: pokemons,
          showAnswer: false,
          showPokemon: false,
        };
      },
    });

    const picture = wrapper.find("pokemon-picture-stub");
    const options = wrapper.find("pokemon-options-stub");

    expect(picture.exists()).toBeTruthy();
    expect(options.exists()).toBeTruthy();
    expect(picture.attributes("pokemonid")).toBe("1");
    expect(options.attributes("pokemons")).toBeTruthy();
  });
});
