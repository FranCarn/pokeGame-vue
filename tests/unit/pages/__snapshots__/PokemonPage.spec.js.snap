// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage tests should match with snapshot 1`] = `<h1>Loading...</h1>`;

exports[`PokemonPage tests should match with snapshot when pokemons exists 1`] = `
<div>
  <h1>Who's that pokemon?</h1>
  <pokemon-picture-stub pokemonid="1" showpokemon="false"></pokemon-picture-stub>
  <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-stub>
  <!--v-if-->
</div>
`;
